spec/job_template_spec.rb

```diff
@@ -16,14 +16,25 @@
     let(:json) { {'extra_vars' => "{\"instance_ids\":[\"i-999c\"],\"state\":\"absent\",\"subnet_id\":\"subnet-887\"}"} }
     let(:post_result_body) { {:job => 1} }
     let(:json_with_limit) { json.merge('limit' => 'machine_name') }
+    let(:response_double) { instance_double("Faraday::Response", :body => post_result_body.to_json) }
 
     it "runs an existing job template with a limit" do
-      launch_url      = "#{raw_instance["url"]}launch/"
-      response_double = instance_double("Faraday::Response", :body => post_result_body.to_json)
+      launch_url = "#{raw_instance["url"]}launch/"
+      instance_data = raw_instance.merge('ask_limit_on_launch' => true)
+
       expect_any_instance_of(AnsibleTowerClient::Collection).to receive(:find).with(post_result_body[:job])
       expect(api).to receive(:post).with(launch_url, json_with_limit).and_return(response_double)
 
-      described_class.new(api, raw_instance).launch(json_with_limit)
+      expect { described_class.new(api, instance_data).launch(json_with_limit) }.to_not raise_error
```

Comment:
No need to specifically assert `to_not raise_error`. The test will fail if the method really raises an error.

Reply:
I know. But I prefer to use meaningful and semantically correct statements. If we test something - we should test (e.g. `expect` something), not just run and see what is going on.

Reply:
Not a big deal here. You can leave it this way. We do have `expect` in line 25 and 26.